### PR TITLE
chore(docs): Update+add documentation

### PR DESCRIPTION
**Pull request by im-coder-lg**
For now, it's just the first commit, I'll continue soon, maybe tomorrow.

### CODE REVIEW

**Comment by im-coder-lg** [comment]
A new user left a comment. This user must be approved by a Netlify team owner before comments can be displayed.

[Approve this user](https://app.netlify.com/teams/sumeshir26/reviewers)

**Comment by im-coder-lg** [comment]
Ah that's me, can you approve me as a reviewer?

File: docs/docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ python3 -m pip install -r requirements.txt
 python3 -m pip install -r windows-requirements.txt
 ```
 After following the series of commands, open your favourite text editor. We recommend that you use [Visual Studio Code](https://code.visualstudio.com). Make your changes and before committing, go to the status bar. There will be a place showing branches and the text `master`. Click that, click `New branch` and name your branch, such as `patch-1` or so. Make sure it correlates to your feature, if it's a patch, name it as a patch, if it's a feature, name the feature. Next, go to the Version Control tab from the sidebar and commit the file, _semantically_. We recommend you to use semantic commits since that is more concise.
-Commit your changes to that new branch, [open a pull request](https://github.com/sumeshir26/TimerX/pulls) and someone will review it ASAP.
+Commit your changes to that new branch, [open a pull request](https://github.com/sumeshir26/TimerX/compare) (click `Compare across forks`) and someone will review it ASAP.
```
Comment (im-coder-lg) [inline docs/docs/CONTRIBUTING.md:20]: :+1:

**Comment by not-nef** [comment]
Yo i can commit cool

**Comment by im-coder-lg** [comment]
I'll manage from here, can you just supervise?

**Comment by im-coder-lg** [comment]
No offense, I need some help frankly.

**Comment by im-coder-lg** [comment]
> Yo i can commit cool

Collaborator rights :sunglasses:

**Comment by im-coder-lg** [comment]
Renamed the title semantically, how's that?

**Comment by not-nef** [comment]
semantically?

**Comment by im-coder-lg** [comment]
The `cz c` way? It's semantic.

**Comment by not-nef** [comment]
what does it mean

**Comment by im-coder-lg** [comment]
If you compare `Add my own image to my blog` and `feat(blog): my own image`, you will say the second one is better than the first. This is semantic committing.
Also @sumeshir26 thanks for approving me as a reviewer.

**Comment by im-coder-lg** [comment]
Is the contributing guide fine? Any changes to add?
Link: https://deploy-preview-42--timerx-app.netlify.app/docs/contributing
<details><summary>Browser metadata</summary>





```
Path:      /docs/contributing
Browser:   Chrome 96.0.4664.137 on Windows 7
Viewport:  1269 x 655 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fcontributing&os=Windows&os_version=7&browser=Chrome&browser_version=96.0&resolution=1269x655&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/contributing) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dafabbd3324300075bb809&commentID=61dba45fd94b171b9bc2f659&resolution=resolved)

**Comment by sumeshir26** [comment]
Can we remove this: https://deploy-preview-42--timerx-app.netlify.app/docs/intro?

**Comment by im-coder-lg** [comment]
No, that's the landing page. We can use it as a welcome page, right>
<details><summary>Browser metadata</summary>





```
Path:      /docs/intro
Browser:   Chrome 96.0.4664.137 on Windows 7
Viewport:  1269 x 655 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fintro&os=Windows&os_version=7&browser=Chrome&browser_version=96.0&resolution=1269x655&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/intro) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dafabbd3324300075bb809&commentID=61dba59ccefee0ff083c2d95&resolution=resolved)

**Comment by im-coder-lg** [comment]
Browser metadata is irritating. Also, I'll edit the intro page, can you add the homepage link to the README and the repo description?

**Comment by sumeshir26** [comment]
Ok

**Comment by im-coder-lg** [comment]
Any ideas on the intro page? Need some text that's why.

**Comment by im-coder-lg** [comment]
Can anyone test c288b78? Also, any other ideas?
<details><summary>Browser metadata</summary>





```
Path:      /docs/compile/
Browser:   Firefox 95.0 on Windows 7
Viewport:  1304 x 683 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fcompile%2F&os=Windows&os_version=7&browser=Firefox&browser_version=95.0&resolution=1304x683&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/compile/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dbbb5a2adf6d00079e1df5&commentID=61dbbd1bd8cd1288935ee0b5&resolution=resolved)

**Comment by im-coder-lg** [comment]
I think I will edit the homepage a bit.

**Comment by sumeshir26** [comment]
@im-coder-lg Look at my review

**Comment by im-coder-lg** [comment]
Where's it?
<details><summary>Browser metadata</summary>





```
Path:      /docs/compile
Browser:   Chrome 97.0.4692.71 on Windows 7
Viewport:  1304 x 665 @1x
Language:  en-GB
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fcompile&os=Windows&os_version=7&browser=Chrome&browser_version=97.0&resolution=1304x665&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/compile) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dbbb5a2adf6d00079e1df5&commentID=61dbc24dd8cd1291585edff2&resolution=resolved)

**Comment by sumeshir26** [comment]
Its right above this comment: 
image

**Comment by im-coder-lg** [comment]
Right above https://github.com/sumeshir26/TimerX/pull/42#issuecomment-1008549823, there is https://github.com/sumeshir26/TimerX/pull/42#issuecomment-1008544552. Try re-commenting that.

**Comment by im-coder-lg** [comment]
I'll be offline for now, it's school. I'll be back by 2:00 PM IST.

**Comment by im-coder-lg** [comment]
I'm back! Now, I will try adding some screenshots to the homepage. @sumeshir26 when you are free, can you give me a desktop screenshot containing TimerX's main window, the configurator window and the settings window? I'll add them to the homepage.

**Comment by sumeshir26** [comment]
I feel like the current image on the README looks nice, it looks cluttered if you open all the windows

**Comment by sumeshir26** [comment]
Also, can repl.it be added to the contributing guide?

**Comment by im-coder-lg** [comment]
> Also, can repl.it be added to the contributing guide?

I'll add that now!

File: docs/docs/compile.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```batch
+REM for making a n unzipped executable package
+python3 setup.py bdist_win
+REM for making an installer:
+python3 setup.py bdist_msi
+```
+If you use Anaconda3:
+```batch
+REM if you want the unzipped package
+python setup.py bdist_win
```
Comment (sumeshir26) [inline docs/docs/compile.md:41]:
```suggestion
python setup.py bdist_msi
```

**Comment by im-coder-lg** [comment]
> I feel like the current image on the README looks nice, it looks cluttered if you open all the windows

It's for the docs homepage. Or, I think the best thing to do is, you take a windowed screenshot of TimerX and send it here. I'll add it to `docs/src/pages/HomepageFeatures.js`.

**Review by sumeshir26 — CHANGES_REQUESTED** [review]

File: docs/docs/compile.md (hunk 2 above)
Comment (im-coder-lg) [reply docs/docs/compile.md:41]: We could create standalone executables, right?

Comment (sumeshir26) [reply docs/docs/compile.md:41]: Sorry typo, now fixed

Comment (im-coder-lg) [reply docs/docs/compile.md:41]: Oh that's `bdist_exe`!

Comment (im-coder-lg) [reply docs/docs/compile.md]: Added!

File: docs/docs/compile.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+REM or if you use Anaconda3,
+pip install -r windows-requirements.txt
+```
+
+## Step 3 {#compile}
+After you install the required packages, execute these commands in TimerX's root directory:
+### Windows compiling
+```batch
+REM for making a n unzipped executable package
+python3 setup.py bdist_win
```
Comment (sumeshir26) [inline docs/docs/compile.md:34]:
```suggestion
python3 setup.py bdist_exe
```

**Comment by im-coder-lg** [comment]
Just one more thing to add, the screenshot.

File: docs/docs/compile.md (hunk 3 above)
Comment (im-coder-lg) [reply docs/docs/compile.md:34]: We need to update the same on Anaconda3's part.

**Comment by im-coder-lg** [comment]
Also, can we try making a triage table for TimerX? It might be clarified and GitHub provides this, unfortunately you need to make a project for your account globally. I will try finding some solution to this though, that would increase productivity.

**Comment by sumeshir26** [comment]
It alreaday has one: https://github.com/sumeshir26/TimerX/projects/1

**Comment by im-coder-lg** [comment]
But that doesn't have tables, that's why I asked. It shows a board, not a table-like view. I mean that it has to look like an Excel spreadsheet.

**Comment by im-coder-lg** [comment]
[Trello](https://trello.com) offers a GitHub integration as well as tabled organizing of cards. But I think we could use something that's already on GitHub.

**Comment by im-coder-lg** [comment]
I removed the old review, it told tat it was pending, so dismissed it with "Resolved".

**Comment by im-coder-lg** [comment]
Wait, `replit` guide needed!

**Comment by im-coder-lg** [comment]
Hey how do we add custom Unicode emojis to Chrome? Becuase I hopped onto a live [BrowserStack](https://live.browserstack.com/dashboard#os=OS+X&os_version=Monterey&browser=Safari&browser_version=15.1&zoom_to_fit=true&full_screen=true&url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2F&speed=1) device running macOS Monterey, Safari 15.1 and they allowed the emoji.
<details><summary>Browser metadata</summary>





```
Path:      /docs/contributing#replit
Browser:   Chrome 97.0.4692.71 on Windows 7
Viewport:  1304 x 665 @1x
Language:  en-GB
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fcontributing%23replit&os=Windows&os_version=7&browser=Chrome&browser_version=97.0&resolution=1304x665&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/contributing#replit) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dc4d1d803b61000814c25c&commentID=61dcf5be6d3fb8794c2e1ca4&resolution=resolved)

**Comment by im-coder-lg** [comment]
Oh, it's globally for macOS.
<details><summary>Browser metadata</summary>





```
Path:      /docs/contributing#replit
Browser:   Chrome 97.0.4692.71 on Windows 7
Viewport:  1304 x 665 @1x
Language:  en-GB
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-42--timerx-app.netlify.app%2Fdocs%2Fcontributing%23replit&os=Windows&os_version=7&browser=Chrome&browser_version=97.0&resolution=1304x665&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-42--timerx-app.netlify.app/docs/contributing#replit) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dc4d1d803b61000814c25c&commentID=61dcf5e0e18b2e6cb88912f8&resolution=resolved)

**Comment by sumeshir26** [comment]
@im-coder-lg Did you add the screenshot?

**Comment by im-coder-lg** [comment]
> Did you add the screenshot?

No, I will add it later. It seems like we all have a busy schedule. I will be busy till lunchtime.

**Comment by im-coder-lg** [comment]
Merge this PR after my commit on the master branch. CI fails and I know why.

**Comment by im-coder-lg** [comment]
Don't, I need to update docs.

**Comment by im-coder-lg** [comment]
Fixed, ready.

**Comment by im-coder-lg** [comment]
Ready for merge!

**Pull request merged**